backend/pkg/handler/groupHandler.go

```diff
@@ -7,103 +7,106 @@ import (
+	groupRepo        *repository.GroupRepository
+	groupMemberRepo  *repository.GroupMemberRepository
+	sessionRepo      *repository.SessionRepository
+	notificationRepo *repository.NotificationRepository
 }
 
 func NewGroupHandler(groupRepo *repository.GroupRepository, sessionRepo *repository.SessionRepository) *GroupHandler {
-    return &GroupHandler{groupRepo: groupRepo, sessionRepo: sessionRepo}
+	return &GroupHandler{groupRepo: groupRepo, sessionRepo: sessionRepo}
 }
 
 // Group Handlers
 func (h *GroupHandler) GetAllGroupsHandler(w http.ResponseWriter, r *http.Request) {
-    // logic for getting all groups
-    groups, err := h.groupRepo.GetAllGroups()
-    if err != nil {
-        http.Error(w, "Failed to get groups: "+err.Error(), http.StatusInternalServerError)
-        return
-    }
-    w.Header().Set("Content-Type", "application/json")
-    json.NewEncoder(w).Encode(groups)
+	// logic for getting all groups
+	groups, err := h.groupRepo.GetAllGroups()
+	if err != nil {
+		http.Error(w, "Failed to get groups: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(groups)
 }
 
 func (h *GroupHandler) CreateGroupHandler(w http.ResponseWriter, r *http.Request) {
-    // logic for creating a group
-    var newGroup model.Group
-    err := json.NewDecoder(r.Body).Decode(&newGroup)
-    if err != nil {
-        http.Error(w, "Failed to decode request body: "+err.Error(), http.StatusBadRequest)
-        return
-    }
-    // TODO: check if group with title already exists IN FRONTEND
-    userID, err := h.sessionRepo.GetUserIDFromSessionToken(util.GetSessionToken(r))
-    if err != nil {
-        http.Error(w, "Error confirming authentication: " + err.Error(), http.StatusInternalServerError)
-        return
-    }
-    newGroup.CreatorId = userID
-    // creating the group in db
-    _, err = h.groupRepo.CreateGroup(newGroup)
-    if err != nil {
-        http.Error(w, "Failed to create group: "+err.Error(), http.StatusInternalServerError)
-        return
-    }
-    w.WriteHeader(http.StatusCreated)
+	// logic for creating a group
+	var newGroup model.Group
+	err := json.NewDecoder(r.Body).Decode(&newGroup)
+	if err != nil {
+		http.Error(w, "Failed to decode request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	// TODO: check if group with title already exists IN FRONTEND
+	userID, err := h.sessionRepo.GetUserIDFromSessionToken(util.GetSessionToken(r))
+	if err != nil {
+		http.Error(w, "Error confirming authentication: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	newGroup.CreatorId = userID
+	// creating the group in db
+	_, err = h.groupRepo.CreateGroup(newGroup)
+	if err != nil {
+		http.Error(w, "Failed to create group: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusCreated)
 
 }
 
 func (h *GroupHandler) GetGroupByIDHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
should check if the authenticated user has requested or has been invited to join the group